nixpkgs_review/nix.py

```diff
@@ -297,7 +297,7 @@ def nix_build(
 def write_shell_expression(
     filename: Path, attrs: list[str], system: str, nixpkgs_config: Path
 ) -> None:
-    with open(filename, "w+") as f:
+    with open(filename, "w+", encoding="utf-8") as f:
```

Comment:
What happens if you don't specify encoding?

Reply:
https://pylint.pycqa.org/en/latest/user_guide/messages/warning/unspecified-encoding.html

> It is better to specify an encoding when opening documents. Using the system default implicitly can create problems on other operating systems. See https://peps.python.org/pep-0597/